src/indicators/StateIndicator.js

```diff
@@ -1,18 +1,42 @@
 import React, { useEffect } from "react";
+import { setStateDependingRandomNum } from "../setStateIndicator/randomNum";
+import { setStateDependingFoodPercent } from "../setStateIndicator/stateDependingFoodPercent";
+import { useTimeout } from "usehooks-ts";
 
 export const StateIndicator = ({
   stateIndicator,
   foodPercent,
   setStateIndicator,
+  buttonTreatDisabled,
+  isRecovered,
+  age,
+  maxAge,
 }) => {
   useEffect(() => {
-    if (foodPercent < 35) {
-      setStateIndicator("hungry");
-      if (foodPercent === 0) setStateIndicator("dead");
-    } else {
-      setStateIndicator("fine");
-    }
+    setStateDependingFoodPercent(foodPercent, setStateIndicator);
   }, [foodPercent]);
+
+  useEffect(() => {
+    if (age > maxAge) setStateIndicator("dead");
+  }, [age]);
+
+  const maxNumForGetRandomNumber = 2;
+
+  useEffect(() => {
+    if (stateIndicator === "fine" || stateIndicator === "hungry") {
+      if (!isRecovered) {
+        setStateDependingRandomNum(maxNumForGetRandomNumber, setStateIndicator);
+      }
+    }
+  }, [stateIndicator]);
+
+  useTimeout(
+    () => {
+      if (stateIndicator === "sick") setStateIndicator("dead");
+    },
+    buttonTreatDisabled ? null : 15000
```

Comment:
should be a variable instead of 15000, like `SICK_PET_TIME_TILL_DEATH_MS`

Reply:
fixed